pkg/tarsum/versioning.go

```diff
@@ -119,9 +119,20 @@ func v1TarHeaderSelect(h *tar.Header) (orderedHeaders [][2]string) {
 	var xattrs [][2]string
 	for k, v := range h.PAXRecords {
 		if xattr, ok := strings.CutPrefix(k, paxSchilyXattr); ok {
+			// h.Xattrs keys take precedence over h.PAXRecords keys, like
+			// archive/tar does when writing.
```

Comment:
I was considering if we had to leave a link to the stdlib code here, but I guess it'd not be too difficult to find.